### PR TITLE
Fambam 541 MessagingInput accessibility and translation support

### PR DESCRIPTION
**Pull request by adamvictorclever**
**Jira:** https://clever.atlassian.net/browse/FAMBAM-541 (see ticket for design instructions/mock)

**Overview:** This PR makes a few changes to make the MessagingInput more accessible and also support translations in the FamPo.

The first commit fixes some accessibility support for `TextArea`. Previously, the underlying `textarea` would never have an `id`, which meant that labels could never properly be associated with them. Previously we used `htmlFor` with the `name` but, I don't believe that actually works. 

Instead, we make a similar change to the one Julie recently made and support using the `name` as the `id` by default. I also added support for explicitly setting an `id` since `name` and `id` do have slightly different requirements (`id` needs to be unique, `name` doesn't). I don't think there will be many cases where explicitly setting an `id` is necessary though.

We also make a change to only include the `TextArea's` internal `label` if there's any text to display. This avoids an issue where the accessibility checker complained about the underlying `textarea` in `MessagingInput` having two labels.

The second commit makes the accessibility changes for `MessagingInput` by adding an external `label` and removing the placeholder text.

The third commit makes it possible to override the text used for the label and send button. This is a temporary measure to be used in the FamPo so that we can pass translations through. In the future, we'd like to support translations inside `clever-components` so that passing them through isn't necessary.

**Screenshots/GIFs:**
Screen Shot 2020-09-30 at 1 01 20 PM

**Testing:**
- [ ] Unit tests
- Manual tests:
  - [ ] Chrome
  - [ ] Safari
  - [ ] IE10

**Roll Out:**
- Before merging:
  - [ ] Updated docs
  - [X] Bumped version in `package.json`
    - Breaking change?
      - If it is a beta component run `npm version minor`
      - If the component is not in beta run `npm version major`
    - New component or backward-compatible component feature change? Run `npm version minor`
    - Only changing documentation? All good. Skip this step.
  - After creating a new component, make sure to add it to the Components List in `ComponentsView.jsx`. To do so:
    - [ ] Add an entry in `ComponentsView.componentsToDisplay` using this template:
      ```
      {
        componentLink: "<COMPONENT LINK>",
        componentImg: "<COMPONENT LINK>.png",
        componentName: "<COMPONENT NAME>",
        componentImgAlt: "A <COMPONENT NAME> component",
      },
      ```
    - [ ] Add a screenshot of the component in `docs/assets/img` with the format `<COMPONENT LINK>.png`
- After merging:
  - [ ] Deployed updated docs (`make deploy-docs`)
  - [ ] Posted in #eng if I made a breaking change to a beta component

### CODE REVIEW

File: src/TextArea/TextArea.tsx [hunk 1]
```diff
@@ -251,9 +259,11 @@ export class TextArea extends React.Component<Props, State> {
         )}
       >
         <div className="TextArea--infoRow">
-          <label className="TextArea--label" htmlFor={this.props.name}>
-            {this.props.label}
-          </label>
+          {this.props.label && (
```
Comment (arsalansufi) [inline src/TextArea/TextArea.tsx:262]: I wonder how many TextArea's exist today with neither a `label` prop specified nor a manually configured label (like in the MessagingInput). It's probably very few. But worth noting that those may end up completely label-less after this is rolled out and imported in repos.

One alternative would be to expose a `noLabel` prop, though I'm not a huge fan. I might just do a quick pass through launchpad's TextArea's to make sure all specify a `label` prop. I know FamPo is okay.

**Review by arsalansufi — COMMENTED** [review]
This looks great! I'll let Julie give the final LGTM

File: src/TextArea/TextArea.tsx (hunk 1 above)
Comment (arsalansufi) [reply src/TextArea/TextArea.tsx:262]: Okay LOL, there are quite a few in launchpad. Literally the first two that I looked at haha.

- https://github.com/Clever/launchpad/blob/03be51877180c72625a97be086e7ec39ab9fbd46/src/ui/components/ClassroomActivityExperiment/ClassroomActivityExperiment.tsx#L90-L100
- https://github.com/Clever/launchpad/blob/03be51877180c72625a97be086e7ec39ab9fbd46/src/ui/components/CollectionPageHeader/EditTeacherAnnouncementModal.tsx#L80-L89

Comment (adamvictorclever) [reply src/TextArea/TextArea.tsx:262]: Hmm, I think I'm not understanding how this change makes those cases any worse. It is true that there may now be some cases where the `<label>` is no longer included, but is there any benefit to rendering a label with no text? It's not clear to me what makes that preferable over no label.

Comment (arsalansufi) [reply src/TextArea/TextArea.tsx:262]: Ah of course 🤦‍♂️

You're absolutely right. Missed the piece that we're just rendering an empty label for those today.

Ignore me :hide:

**Review by arsalansufi — APPROVED** [review]
I know Julie hasn't had a chance to look at this yet, but LGTM!

File: src/TextArea/TextArea.tsx (hunk 1 above)
Comment (adamvictorclever) [reply src/TextArea/TextArea.tsx:262]: Haha no worries, glad we're on the same page 😄

**Pull request merged**